providers/squadcast/squadcast_provider.go

```diff
@@ -0,0 +1,179 @@
+package squadcast
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/zclconf/go-cty/cty"
+
+	"github.com/GoogleCloudPlatform/terraformer/terraformutils"
+)
+
+type SquadcastProvider struct {
+	terraformutils.Provider
+	accesstoken  string
+	refreshtoken string
+	region       string
+	teamName     string
+}
+
+type AccessToken struct {
+	Type         string `json:"type"`
+	AccessToken  string `json:"access_token"`
+	IssuedAt     int64  `json:"issued_at"`
+	ExpiresAt    int64  `json:"expires_at"`
+	RefreshToken string `json:"refresh_token"`
+}
+
+// Meta holds the status of the request information
+type Meta struct {
+	Meta AppError `json:"meta,omitempty"`
+}
+
+type AppError struct {
+	Status  int    `json:"status"`
+	Message string `json:"error_message,omitempty"`
+}
+
+const (
+	UserAgent = "terraform-provider-squadcast"
+)
+
+func (p *SquadcastProvider) Init(args []string) error {
+
+	if refreshToken := os.Getenv("SQUADCAST_REFRESH_TOKEN"); refreshToken != "" {
+		p.refreshtoken = os.Getenv("SQUADCAST_REFRESH_TOKEN")
+	}
+	if args[0] != "" {
+		p.refreshtoken = args[0]
+	}
+	if p.refreshtoken == "" {
+		return errors.New("required refresh Token missing")
+	}
+
+	if region := os.Getenv("SQUADCAST_REGION"); region != "" {
+		p.region = os.Getenv("SQUADCAST_REGION")
+	}
+	if args[1] == "" {
+		return errors.New("required region missing")
+	}
+	p.region = args[1]
+
+	if args[2] != "" {
+		p.teamName = args[2]
+	}
+
+	p.GetAccessToken()
+	return nil
+}
+
+func (p *SquadcastProvider) InitService(serviceName string, verbose bool) error {
+	var isSupported bool
+	if _, isSupported = p.GetSupportedService()[serviceName]; !isSupported {
+		return errors.New(p.GetName() + ": " + serviceName + " not supported service")
+	}
+	p.Service = p.GetSupportedService()[serviceName]
+	p.Service.SetName(serviceName)
+	p.Service.SetVerbose(verbose)
+	p.Service.SetProviderName(p.GetName())
+	p.Service.SetArgs(map[string]interface{}{
+		"access_token":  p.accesstoken,
+		"refresh_token": p.refreshtoken,
+		"region":        p.region,
+		"team_name":     p.teamName,
+	})
+	return nil
+}
+
+func (p *SquadcastProvider) GetConfig() cty.Value {
+	return cty.ObjectVal(map[string]cty.Value{
+		"region":        cty.StringVal(p.region),
+		"refresh_token": cty.StringVal(p.refreshtoken),
+	})
+}
+
+func (p *SquadcastProvider) GetProviderData(...string) map[string]interface{} {
+	return map[string]interface{}{
+		"provider": map[string]interface{}{
+			"squadcast": map[string]interface{}{
+				"region": p.region,
```

Comment:
just wondering, can't we set `source` block here itself? since it's just map[string]

Reply:
we can't do that actually, it'll throw an error
![image](https://user-images.githubusercontent.com/72292326/187365811-5298223b-42ca-48b6-b5d8-4e4abbe5a34e.png)

Also https://www.terraform.io/language/providers/configuration#version-deprecated

Reply:
sure